core/src/main/kotlin/run/qontract/core/utilities/ContractSource.kt

```diff
@@ -58,14 +57,14 @@ data class GitRepo(val gitRepositoryURL: String, override val testContracts: Lis
     }
 }
 
-data class GitMonoRepo(override val testContracts: List<String>, override val stubContracts: List<String>) : ContractSource() {
+data class GitMonoRepo(override val testContracts: List<String>, override var stubContracts: List<String>) : ContractSource() {
```

Comment:
I would like to avoid using mutable state as far as possible. It should be fairly easy to do. We can pair on this.

Reply:
refactored to add a new method to the ContractSource - loadContracts, which keeps the ContractSource immutable.